src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/AuthenticationTests.cs

```diff
@@ -185,6 +185,30 @@ public async Task ApiKeyAuthenticationSchemeTest()
             Assert.NotNull(processes);
         }
 
+        /// <summary>
+        /// Tests that API key authentication can be configured correctly and
+        /// that the key can be rotated wihtout shutting down dotnet-monitor.
+        /// </summary>
+        [Fact]
+        public async Task TempApiKeyAuthenticationSchemeTest()
```

Comment:
Could you add another test that enables the temp API key usage and custom API key at the same time and validate that the custom API key will 401 whereas the temp API key will 200?